app/assets/javascripts/exchanges/carriers.js

```diff
@@ -0,0 +1,70 @@
+$(document).on('turbolinks:load', function() {
+    $('.filter-form button[type="submit"]').on('click', function(event) {
+        event.preventDefault();
+
+        filterPlans();
+    });
+
+    $('#clear-filters').on('click', function() {
+        $('.filter-form input[type="checkbox"]').prop('checked', false);
+        $('.filter-form input[type="text"]').val('');
+
+        filterPlans();
+    });
+
+    function filterPlans() {
+        const planTypes = [];
+        const pvpRatingAreas = [];
+        const metalLevels = [];
+        const searchPlan = $('input[name="search"]').val().trim().toLowerCase();
+
+        $('input[name="plan_type[]"]:checked').each(function() {
+            planTypes.push($(this).val().toLowerCase());
+        });
+
+        $('input[name="pvp_rating_areas[]"]:checked').each(function() {
+            pvpRatingAreas.push($(this).val().toLowerCase());
+        });
+
+        $('input[name="metal_level[]"]:checked').each(function() {
+            metalLevels.push($(this).val().toLowerCase());
+        });
+
+        let visiblePlansCount = 0;
+
+        $('tbody tr').each(function() {
+            const planType = $(this).data('plan-type').toString().toLowerCase();
+            const pvpAreas = $(this).data('pvp-areas').toString().toLowerCase();
+            const metalLevel = $(this).data('metal-level').toLowerCase();
+            const planId = $(this).data('plan-id').toLowerCase();
+            const planName = $(this).data('plan-name').toLowerCase();
+
+            let showRow = true;
+
+            if (planTypes.length > 0 && !planTypes.some(pt => planType.includes(pt))) {
+                showRow = false;
+            }
+
+            if (pvpRatingAreas.length > 0 && !pvpRatingAreas.includes(pvpAreas)) {
```

Comment:
I think this is comparing an array of (single) areas from the filter with a comma-joined series of areas in a string from a given row... Right? So, for filter on areas 1,2 and 3 and a given row with `"2,44"`, the comparison would be `... && !["1","2","3"].includes("2,44") ...`, which would not match. I'm not sure what the logic should be exactly (i.e., do you need to just match one rating area to show the row?), but it's seems like this will fail to match if any plan has >1 rating area. Something to verify?